### PR TITLE
libpng: 1.6.35 -> 1.6.36, license v2

### PR DESCRIPTION
**Pull request by dtzWill**
###### Motivation for this change

https://sourceforge.net/p/png-mng/mailman/message/36483059/

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: lib/licenses.nix [hunk 1]
```diff
@@ -442,6 +442,11 @@ lib.mapAttrs (n: v: v // { shortName = n; }) rec {
     fullName = "libpng License";
   };
 
+  libpng2 = {
+    fullName = "libpng License v2"; # 1.6.36+
+    inherit (libpng) url;
```
Comment (c0bw3b) [inline lib/licenses.nix]: Humm no that will point to the SPDX url of the previous/old license.
I think `url` should rather be https://sourceforge.net/p/libpng/code/ci/v1.6.36/tree/LICENSE or http://www.libpng.org/pub/png/src/libpng-LICENSE.txt

At least until SPDX adds a new entry or updates the `libpng` entry.

**Comment by c0bw3b** [comment]
So... they do a licensing change in a patch version, to begin with.
And the result is just... :

> The new libpng license comprises the terms and conditions from the zlib
license, and the disclaimer from the Boost license.

Oh come one... Just choose one of the two licenses already defined and accepted as free instead of (not) reinventing the wheel by composing parts of others' licenses...

File: pkgs/development/libraries/libpng/default.nix [hunk 2]
```diff
@@ -25,14 +25,14 @@ in stdenv.mkDerivation rec {
 
   propagatedBuildInputs = [ zlib ];
 
-  doCheck = stdenv.hostPlatform == stdenv.buildPlatform;
+  doCheck = true;
```
Comment (Mic92) [inline pkgs/development/libraries/libpng/default.nix]: This will be still skipped for cross-builds, right?

**Comment by c0bw3b** [comment]
@GrahamcOfBorg eval

**Comment by c0bw3b** [comment]
@GrahamcOfBorg eval

File: pkgs/development/libraries/libpng/default.nix (hunk 2 above)
Comment (dtzWill) [reply pkgs/development/libraries/libpng/default.nix]: Yes, that's my understanding! :)

File: lib/licenses.nix (hunk 1 above)
Comment (dtzWill) [reply lib/licenses.nix]: Well the SPDX page itself links to http://www.libpng.org/pub/png/src/libpng-LICENSE.txt , but you're right the libpng spdx identifier as-is seems incorrect for now.  Bah.

I'll bump to point to the license directly? Coming shortly...

**Comment by dtzWill** [comment]
Okay to merge IMO, any objections?

**Comment by vcunat** [comment]
Bikeshedding: maybe we don't need the definition in `licenses.nix`, as it doesn't seem likely to be reused, but I don't really care which of the two ways we use here.

**Review by vcunat — APPROVED** [review]

**Comment by vcunat** [comment]
Cross-links: https://github.com/glennrp/libpng/issues/266 98be12ab3a

**Pull request merged**